### DISCUSSION TITLE
New Neper command to cut a tessellation by a rounded cube

### DISCUSSION THREAD

**Discussion by rquey** [neperfepx/neper#814] [Show and tell]
Neper (version `4.8.2-1`) now has a `-transform "cut(cube(...))"` option that cuts a tessellation by a cube with rounded corners. This can be used, for instance, to model tensile specimens with flat edges and rounded shoulders:
```
neper -T -n 1000 -morpho gg -morphooptistop "val=1e-2" -domain "cube(1,0.2,4)" -transform "cut(cube(-10,0.25,-10,10,1,3,0.27),cube(0.75,10,-10,10,1,3,0.27))" -reg 1                                                                                          
neper -M n1000-id1.tess -rcl 0.8
neper -V n1000-id1.tess,n1000-id1.msh -dataedgerad "((domtype==1)?0.006:0.004)" -showelt1d all -dataelt1drad "((domtype==1)?0.006:0.004)" -dataelt3dedgerad 0.001 -imagesize 500:1000 -cameraangle 8.5 -showtess 1 -showmesh 0 -print tess -showtess 0 -showmesh 1 -print mesh
convert +append tess.png mesh.png tess-mesh.png
```
![tess-mesh](https://github.com/neperfepx/neper/assets/2659376/453a8167-8c12-4cd9-b63c-1ebbd1e5e162)
As cutting in "experimental" (i.e., not 100% stable), it may fail sometimes. In such cases, increasing `-n` of changing `-id` can help.

**Comment by charanjo** [comment]
Hi, Can a cylindrical torsion specimen be created like this? 
![Picture1](https://github.com/user-attachments/assets/2a788335-d9c0-476f-ba22-7a273ee1b028)

**Comment by Zdmire** [comment]
Hi, thanks for your sharing. However, i encountered a problem when i tried to cut a tessellation by a round cube in 2D. There were always errors, so following your instructions, I increased the number of -n and was able to successfully divide it. However, a new problem emerged. The grains at the division boundary became distorted and formed very large grains. How can we improve this step?. I haven't been able to solve this kind of cutting problem in 2D for several days.  @rquey Could you help me?

neper -T -n 2000 -morpho gg -morphooptistop "val=1e-2" -domain "square(1,4)" -transform "cut(cube(0.45,0.55
,-2,2,-3,3,0.27))" -dim 2 -reg 1

neper -V n2000-id1.tess -datacellcol id -imagesize 1200:1200 -print img

![image](https://github.com/user-attachments/assets/346c6ba1-8eea-49ff-8dd8-36644cfe7c9d)

**Reply by rquey** [reply]
This is a limit case (radius of curvature < cell size) that does not work properly.
